test/Tensor_test.jl

```diff
@@ -41,6 +41,21 @@
 
         @test tensor ∈ Set([tensor])
         @test zeros(size(tensor)...) ∉ Set([tensor])
+
+        @test tensor == permutedims(tensor, (3, 1, 2))
+    end
+
+    @testset "isapprox" begin
+        tensor = Tensor(zeros(2, 2, 2), (:i, :j, :k))
```

Comment:
Can you check out with random number rather than with zeros? If all elements are equal, we cannot check correctly that permutations have been performed correctly.